### PR TITLE
POC using Lexical Editor

### PR DESCRIPTION
**Pull request by cguedes**
This PR is the POC for using Lexical within the evaluation of text editors being discussed in #4 

POC with 3 panels:
 - Left Panel: References list
 - Center Panel: Lexical editor with rich-text
 - Right Panel : AI interactions

Selections of text are listed in the right panel, and clicks on a reference (left panel) add the text (ex: Ref 3) to the selected/cursor area in the editor (central panel).

image

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -5,7 +5,7 @@
     <meta name="viewport" content="width=device-width, initial-scale=1.0" />
   </head>
 
-  <body>
+  <body style="margin: 0">
```
Comment (sergioramos) [inline index.html:8]: You can have a root `css` file with some resets.

**Comment by hammer** [comment]
Hey small development process that I like to follow: I prefer to have an issue open before a PR is opened. @cguedes could you open an issue for a Lexical prototype so that we can discuss high-level thoughts there and code-level thoughts here in the PR?

**Comment by hammer** [comment]
Up to you!

**Comment by danvk** [comment]
Another small point on development process: it'll be easier for everyone if we create branches on this repo (rather than  fork) for PRs. I think everyone should have push rights? Or can easily be given them.

**Comment by hammer** [comment]
Wow a lot of features already implemented! A few thoughts:

- We can resize components now. How easy is it to collapse them entirely, or re-arrange them?
- What is the difference between "Uploads" and "References"? Do we want them to be separate?
- If we wanted to have multiple documents, is it easy to add them as tabs as in VS Code? Long term this creates questions about the scope of references and AI interactions. Nothing happens if I click "file2.txt", e.g.
- ">" to create a collapsible block behaves weirdly
- "Markdown" toggle does not seem to be able to toggle back.
- Inserting references already, nice! It would be nice to be able to insert from Markdown using the appropriate syntax.
- Vertical alignment of Lexical toolbar is odd, it's covered by the title bar and doesn't show fully even in fll screen mode.

**Comment by cguedes** [comment]
Closing this in favor of https://github.com/refstudio/refstudio/pull/23 that integrates Lexical using the most recent codebase.

**Pull request closed without merging**